File: rss-reader/app/src/main/java/com/abubusoft/kripton/examples/rssreader/service/model/Channel.java
package com.abubusoft.kripton.examples.rssreader.service.model;

import java.util.Date;
import java.util.List;

import com.abubusoft.kripton.android.ColumnType;
import com.abubusoft.kripton.android.annotation.BindSqlColumn;
import com.abubusoft.kripton.android.annotation.BindSqlRelation;
import com.abubusoft.kripton.android.annotation.BindSqlType;
import com.abubusoft.kripton.annotation.Bind;
import com.abubusoft.kripton.annotation.BindAdapter;
import com.abubusoft.kripton.annotation.BindType;

@BindType
@BindSqlType(name="channels")
public class Channel extends Entity {

	public String title;

	@BindSqlColumn(columnType = ColumnType.UNIQUE)
	public String link;

	public String description;
	public String language;
	public String copyright;
	
	@BindAdapter(adapter = DateAdapter.class)
	public Date pubDate;
	
	@BindAdapter(adapter = DateAdapter.class)
	public Date lastBuildDate;
	
	public Image image;

	@BindSqlColumn(parentEntity = RssFeed.class)
	public long rssFeedId;

	@Bind("item")
	@BindSqlRelation(foreignKey = "channelId")
	public List<Article> articles;
}

File: kotlin-rss-reader/app/src/main/java/com/abubusoft/kripton/example/rssreader/service/model/Channel.kt
package com.abubusoft.kripton.example.rssreader.service.model

import com.abubusoft.kripton.android.annotation.BindSqlRelation
import com.abubusoft.kripton.annotation.Bind
import com.abubusoft.kripton.android.annotation.BindSqlColumn
import com.abubusoft.kripton.annotation.BindAdapter
import com.abubusoft.kripton.android.ColumnType
import com.abubusoft.kripton.android.annotation.BindSqlType
import com.abubusoft.kripton.annotation.BindType
import java.util.*

@BindType
@BindSqlType(name = "channels")
data class Channel(
        @BindSqlColumn(columnType = ColumnType.PRIMARY_KEY) var id: Long = 0,
        val title: String? = null,

        @BindSqlColumn(columnType = ColumnType.UNIQUE)
        val link: String? = null,

        val description: String? = null,
        val language: String? = null,
        val copyright: String? = null,

        @BindAdapter(adapter = DateAdapter::class)
        val pubDate: Date? = null,

        @BindAdapter(adapter = DateAdapter::class)
        val lastBuildDate: Date? = null,

        val image: Image? = null,

        @BindSqlColumn(parentEntity = RssFeed::class)
        var rssFeedId: Long = 0,

        @Bind("item")
        @BindSqlRelation(foreignKey = "channelId")
        val articles: List<Article>? = null
)